pkg/reconciler/route/resources/service_test.go

```diff
@@ -151,3 +192,59 @@ func TestNewMakeK8SService(t *testing.T) {
 		}
 	}
 }
+
+func TestMakePlaceholderK8sService(t *testing.T) {
+	target := traffic.RevisionTarget{
+		TrafficTarget: v1beta1.TrafficTarget{
+			Tag: "foo",
+		},
+	}
+
+	ctx := context.Background()
+	cfg := testConfig()
+	ctx = config.ToContext(ctx, cfg)
+
+	service, err := MakeK8sPlaceholderService(ctx, r, target.Tag)
+	expectedMeta := metav1.ObjectMeta{
+		Name:      domains.SubdomainName(r, target.Tag),
+		Namespace: r.Namespace,
+		OwnerReferences: []metav1.OwnerReference{
+			*kmeta.NewControllerRef(r),
+		},
+	}
+	expectedSpec := corev1.ServiceSpec{
+		Type:         corev1.ServiceTypeExternalName,
+		ExternalName: "test-route-foo.test-ns.example.com",
+	}
+
+	if err != nil {
+		t.Errorf("Unexpected error %v returned", err)
+	}
+
+	if diff := cmp.Diff(expectedMeta, service.ObjectMeta); diff != "" {
```

Comment:
I always recommend to use `cmp.Equal` and to compute difference only if the equality failed. Equality is much cheaper and in general the unit tests are passing most of the time, so no need to do more work than necessary.
Also `diff` is a string, so `%s` is more expressive.